### PR TITLE
CMSIS-NN: Add unit test support for fully_connected

### PR DESCRIPTION
**Pull request by mansnils**

### CODE REVIEW

File: CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Generated by generate_test_data.py using TFL version 2.1.0 as reference.
+#define FULLY_CONNECTED_OUT_CH 6
+#define FULLY_CONNECTED_IN_CH 10
+#define FULLY_CONNECTED_INPUT_W 2
+#define FULLY_CONNECTED_INPUT_H 1
+#define FULLY_CONNECTED_FILTER_X 2
+#define FULLY_CONNECTED_FILTER_Y 1
+#define FULLY_CONNECTED_STRIDE_X 1
+#define FULLY_CONNECTED_STRIDE_Y 1
+#define FULLY_CONNECTED_PAD_X 0
```
Comment (felix-johnny) [inline CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h:30]: padding and stride are not part of Fully connected. So, we can remove them..

File: CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+
+#pragma once
+// Generated by generate_test_data.py using TFL version 2.1.0 as reference.
+#define FULLY_CONNECTED_OUT_CH 6
+#define FULLY_CONNECTED_IN_CH 10
+#define FULLY_CONNECTED_INPUT_W 2
+#define FULLY_CONNECTED_INPUT_H 1
+#define FULLY_CONNECTED_FILTER_X 2
```
Comment (felix-johnny) [inline CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h:26]: filter x,y and output x, y are unused parameters. So, please remove them. We have a 2D input and a 2D filter.. arm_nnfunctions.h describe what parameters are used.

File: CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h (hunk 1 above)
Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h]: Yes, it is done.

File: CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h (hunk 2 above)
Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/TestData/fully_connected/config_data.h]: Done

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_fully_connected_s8/test_arm_fully_connected_s8.c [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const q7_t *output_ref = fully_connected_output_ref;
+  const int32_t output_ref_size = FULLY_CONNECTED_DST_SIZE;
+
+  input_dims.n = FULLY_CONNECTED_INPUT_BATCHES;
+  input_dims.w = FULLY_CONNECTED_INPUT_W;
+  input_dims.h = FULLY_CONNECTED_INPUT_H;
+  input_dims.c = FULLY_CONNECTED_IN_CH;
+  filter_dims.n = FULLY_CONNECTED_ACCUMULATION_DEPTH;
+  filter_dims.c = FULLY_CONNECTED_OUT_CH;
+  output_dims.c = FULLY_CONNECTED_OUT_CH;
```
Comment (felix-johnny) [inline CMSIS/NN/Tests/UnitTest/TestCases/test_arm_fully_connected_s8/test_arm_fully_connected_s8.c:51]: Could you initialize output_dims.n as well? I am comparing the initialization here with what is defined in arm_nnfunctions.h so it stands as a reference for a user.

Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/test_arm_fully_connected_s8/test_arm_fully_connected_s8.c:51]: Yes, will do.

**Pull request merged**